src/io/flutter/view/FlutterView.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.view;
+
+import com.intellij.icons.AllIcons;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.Storage;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.SimpleToolWindowPanel;
+import com.intellij.openapi.wm.ToolWindow;
+import com.intellij.ui.content.Content;
+import com.intellij.ui.content.ContentFactory;
+import com.intellij.ui.content.ContentManager;
+import io.flutter.FlutterInitializer;
+import io.flutter.actions.OpenObservatoryAction;
+import io.flutter.run.daemon.FlutterApp;
+import org.dartlang.vm.service.VmServiceListener;
+import org.dartlang.vm.service.element.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+// TODO(devoncarew): Display an fps graph.
+// TODO(devoncarew): Have a pref setting for opening when starting a debug session.
+
+@com.intellij.openapi.components.State(
+  name = "FlutterView",
+  storages = {@Storage("$WORKSPACE_FILE$")}
+)
+public class FlutterView implements PersistentStateComponent<FlutterView.State>, Disposable {
+  private FlutterView.State state = new FlutterView.State();
```

Comment:
Since you've ensure this is not null in the constructor, it wouldn't hurt to annotate it that way here.